trie.h

```diff
@@ -27,12 +27,13 @@ typedef struct
     uint8_t     completion_len;     // length of completion string
     uint8_t     num_backspaces;     // number of backspaces to send before the completion string
     uint8_t     func_code;          // special function code
+    uint8_t     context_len;
```

Comment:
I believe this should be called `depth` and set equal to the depth that the result was matched in the trie.

You should name a field based on what it IS, not how you plan to use it. The `st_trie_payload` is a struct containing information about a match in the trie. This filed represents how deep into the trie the match was made.

Additionally, calling it `context_len` is even confusing when you use it, because it isn't the length of the context, it's some shorter length equal to the length of the matching portion of the context.